### PR TITLE
Helper APIs to navigate Mo, class Ids, Mo Properties

### PR DESCRIPTION
**Pull request by ragupta-git**
Added API:
`search_class_id`  To find a given class_id exist in meta.(case insensitive)
`get_meta_info`  This gives the meta info for the respective class_id and its associated properties and prints the mo hierarchy tree.
```
from ucsmsdk.ucscoreutils import get_meta_info
meta = get_meta_info(class_id="MeMOryaRRay")

[ComputeBoard]
  |-MemoryArray
     |-FaultInst
     |-MemoryArrayEnvStats
     |  |-MemoryArrayEnvStatsHist
     |-MemoryUnit
        |-FaultInst
        |-MemoryErrorStats
        |-MemoryUnitEnvStats
           |-MemoryUnitEnvStatsHist

ClassId                         MemoryArray
-------                         -----------
xml_attribute                   :memoryArray
rn                              :memarray-[id]
min_version                     :1.0(1e)
access                          :InputOutput
access_privilege                :['read-only']
parents                         :['computeBoard']
children                        :['faultInst', 'memoryArrayEnvStats', 'memoryUnit']

Property                        child_action
--------                        ------------
xml_attribute                   :childAction
field_type                      :string
min_version                     :1.0(1e)
access                          :INTERNAL
min_length                      :None
max_length                      :None
pattern                         :((deleteAll|ignore|deleteNonPresent),){0,2}(deleteAll|ignore|deleteNonPresent){0,1}
value_set                       :[]
range_val                       :[]

Property                        cpu_id
--------                        ------
xml_attribute                   :cpuId
field_type                      :uint
min_version                     :1.0(1e)
access                          :READ_ONLY
min_length                      :None
max_length                      :None
pattern                         :None
value_set                       :[]
range_val                       :[]

Property                        curr_capacity
--------                        -------------
xml_attribute                   :currCapacity
field_type                      :string
min_version                     :1.0(1e)
access                          :READ_ONLY
min_length                      :None
max_length                      :None
pattern                         :None
value_set                       :['unspecified']
range_val                       :['0-4294967295']

:
:
:
:

Property                        voltage
--------                        -------
xml_attribute                   :voltage
field_type                      :string
min_version                     :1.0(1e)
access                          :READ_ONLY
min_length                      :None
max_length                      :None
pattern                         :None
value_set                       :['lower-critical', 'lower-non-critical', 'lower-non-recoverable', 'not-supported', 'ok', 'unknown', 'upper-critical', 'upper-non-critical', 'upper-non-recoverable']
range_val                       :[]
```

`include_prop=False`  will not display the properties.
```
meta = get_meta_info(class_id="MemoryArray", include_prop=False)
```

`show_tree=False` will not display the tree.
```
meta = get_meta_info(class_id="MemoryArray", show_tree=False)
```

```
meta = get_meta_info(class_id="MemoryArray", include_prop=False, show_tree=False)
print("\n-----------------------------------------------------")
print(meta.xml_attribute)
print(meta.parents)
print(meta.children)

ClassId                         MemoryArray
-------                         -----------
xml_attribute                   :memoryArray
rn                              :memarray-[id]
min_version                     :1.0(1e)
access                          :InputOutput
access_privilege                :['read-only']
parents                         :['computeBoard']
children                        :['faultInst', 'memoryArrayEnvStats', 'memoryUnit']

-----------------------------------------------------
memoryArray
['computeBoard']
['faultInst', 'memoryArrayEnvStats', 'memoryUnit']
```

Class Id not present but related class Id present in meta.
```
meta = get_meta_info(class_id="MeM")

2016-03-21 11:32:30,899 - ucs - INFO - "MeM" did not match any available Class Ids.
Related Class Ids are:
----------------------
BiosVfMaximumMemoryBelow4GB
BiosVfMemoryMappedIOAbove4GB
BiosVfSelectMemoryRASConfiguration
ComputeMemoryConfigPolicy
ComputeMemoryConfiguration
ComputeMemoryUnitConstraintDef
EquipmentMemoryUnitCapProvider
EquipmentMemoryUnitDiscoveryModifierDef
FabricVsanMembership
InitiatorMemberEp
LsZoneInitiatorMember
LsZoneTargetMember
MemoryArray
MemoryArrayEnvStats
MemoryArrayEnvStatsHist
MemoryBufferUnit
MemoryBufferUnitEnvStats
MemoryBufferUnitEnvStatsHist
MemoryErrorStats
MemoryQual
MemoryRuntime
MemoryRuntimeHist
MemoryUnit
MemoryUnitEnvStats
MemoryUnitEnvStatsHist
PowerChassisMember
PowerRackUnitMember
StorageVDMemberEp
SwZoneInitiatorMember
SwZoneTargetMember
```

Class Id and no matching Class Id present in meta.
```
meta = get_meta_info(class_id="MeX")

2016-03-21 11:33:39,485 - ucs - INFO - "MeX" did not match any available Class Ids.
```

### CODE REVIEW

**Comment by ragupta-git** [comment]
@vvb: Made include_prop and show_tree as True by default.

**Comment by vvb** [comment]
LGTM

File: ucsmsdk/ucscoreutils.py [hunk 1]
```diff
@@ -483,3 +483,174 @@ def get_naming_props(rn_str, rn_pattern):
...
+
+    if meta_class_id is not None:
+        return meta_class_id
+
+    # if class_id not exists in meta
+    l_class_id = class_id.lower()
+    class_ids = sorted([cid for cid in ucsmeta.MO_CLASS_ID
+                        if re.search(l_class_id, cid, re.IGNORECASE)])
+    if class_ids:
+        log.info("<%s> is not present.\n"
```
Comment (vvb) [inline ucsmsdk/ucscoreutils.py]: @ragupta-git Skip the message here. Just print all the classIds that matched.

File: ucsmsdk/ucscoreutils.py [hunk 2]
```diff
@@ -483,3 +483,174 @@ def get_naming_props(rn_str, rn_pattern):
...
+                 "Related available class_ids are:\n%s" %
+                 (class_id, "\n".join(class_ids)))
+    else:
+        log.info("<%s> is not present." % class_id)
```
Comment (vvb) [inline ucsmsdk/ucscoreutils.py]: @ragupta-git "<%s> did not match any available Class Ids"

File: ucsmsdk/ucscoreutils.py [hunk 3]
```diff
@@ -483,3 +483,174 @@ def get_naming_props(rn_str, rn_pattern):
...
+        meta = get_meta_info(class_id="lsserver", include_prop=False)
+        meta = get_meta_info(class_id="lsserver", show_tree=False)
+
+        print(meta.xml_attribute)
+        print(meta.children)
+        print(meta.props["name"])
+    """
+
+    meta_class_id = search_class_id(class_id)
+    if not meta_class_id:
```
Comment (vvb) [inline ucsmsdk/ucscoreutils.py]: @ragupta-git no need of an exception here, a return is enough. search_class_id throws the necessary msg.

Comment (ragupta-git) [inline ucsmsdk/ucscoreutils.py]: Removed the exception and changed the message.

**Comment by vvb** [comment]
LGTM

**Pull request merged**